### PR TITLE
vscode-extensions: add a few popular extensions

### PR DESCRIPTION
**Pull request by schuelermine**
###### Description of changes

Added a few popular VSCode extensions

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -1774,6 +1774,40 @@ let
 
       ms-vscode.cpptools = callPackage ./cpptools { };
 
+      ms-vscode.cmake-tools = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "cmake-tools";
+          publisher = "ms-vscode";
+          version = "1.13.17";
+          sha256 = "0szgwqlj2cbww7ciykv5za3ldj4kw1xfvrq6bfh4mdq7xdb2k523";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/default.nix:1782]:
```suggestion
          sha256 = "sha256-RF4KTHXnYBfRCD06GYmprwJNkdplyZyk4ioq5Nu+UDM=";
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 197371` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vscode-extensions.ms-vscode.cmake-tools</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>vscode-extensions.ms-toolsai.jupyter-keymap</li>
    <li>vscode-extensions.ms-toolsai.vscode-jupyter-cell-tags</li>
    <li>vscode-extensions.ms-toolsai.vscode-jupyter-slideshow</li>
    <li>vscode-extensions.ms-vscode.hexeditor</li>
    <li>vscode-extensions.ms-vscode.makefile-tools</li>
    <li>vscode-extensions.twxs.cmake</li>
    <li>vscode-extensions.vscjava.vscode-java-dependency</li>
    <li>vscode-extensions.vscjava.vscode-maven</li>
    <li>vscode-extensions.vscjava.vscode-spring-initializr</li>
  </ul>
</details>

**Comment by schuelermine** [comment]
@SuperSandro2000 fixed it

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 1 above)
Comment (schuelermine) [reply pkgs/applications/editors/vscode/extensions/default.nix]: fixed it

**Pull request merged**